src/nomad_simulations/outputs.py

```diff
@@ -15,38 +15,153 @@
+        with the class name of each output property class, e.g., `'ElectronicBandGap'`, `'ElectronicBandStructure'`, etc.
+        """,
+        a_eln=ELNAnnotation(component='StringEditQuantity'),
+    )
+
+    orbitals_state_ref = Quantity(
+        type=OrbitalsState,
+        description="""
+        Reference to the `OrbitalsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    atoms_state_ref = Quantity(
+        type=AtomsState,
+        description="""
+        Reference to the `AtomsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    model_system_ref = Quantity(
+        type=ModelSystem,
+        description="""
+        Reference to the `ModelSystem` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the output property is derived from other output properties. We make
+        the distinction between directly parsed, derived, and post-processing output properties:
+            - Directly parsed: the output property is directly parsed from the simulation output files.
+            - Derived: the output property is derived from other output properties. No extra numerical settings
+                are required to calculate the output property.
+            - Post-processing: the output property is derived from other output properties. Extra numerical settings
+                are required to calculate the output property.
+        """,
+    )
+
+    outputs_ref = Quantity(
+        type=Reference(SectionProxy('BaseOutputs')),
+        description="""
+        Reference to the `BaseOutputs` section from which the output property was derived. This is only
+        relevant if `is_derived` is set to True.
+        """,
+    )
+
+    def check_is_derived(self, is_derived: bool, outputs_ref) -> Optional[bool]:
+        if not is_derived:
+            if outputs_ref is not None:
+                return True
+            return False
+        elif is_derived and outputs_ref is not None:
+            return True
+        return None
+
     def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
-        self.logger = logger
+
+        check_derived = self.check_is_derived(self.is_derived, self.outputs_ref)
+        if check_derived is not None:
+            self.is_derived = check_derived
+        else:
+            logger.error(
+                'A derived output property must have a reference to another `Outputs` section.'
+            )
+            return
+
+
+class Outputs(BaseOutputs):
+    """
+    Output properties of a simulation.
+
+    # ! add more description once we defined the output properties
+    """
+
+    n_scf_steps = Quantity(
```

Comment:
with all these scf-specific quantities, this seems like a more specific output property class. Additionally, it's also a little confusing to me how something like "Number of self-consistent steps to converge the output property" is not a workflow property. This is dependent on the output property value of several individual model systems, correct?

Reply:
It was decided before joining FAIRmat, I think, that all scf steps will belong to the same `SinglePoint` concept, just on the same basis that breaking System into individual entries does not make too much sense from the computational side.

But yes, being very strict, scf steps are a workflow. I will add more description into these quantities and link this with the `SinglePoint` concept in NOMAD. Eventually, these things should be carefully described in the documentation page.

Reply:
I would keep SCF with the same entry, since we consider single-point calculations as an elementary task.

> this seems like a more specific output property class

Agreed. If it uses the same concepts, you can inherit, else keep them separate.

Reply:
I wasn't arguing to split into different entries, but now I am questioning a bit my understanding of the plan. We are planning to keep the current basic structure for Outputs right? Where Outputs is a list with each member corresponding to a single configuration? If this is the case I don't see how `n_scf_steps` makes sense here. Doesn't it need to be stored at a higher level?

My bigger concern is about the specificity of this class. The Outputs class sounds general, but these quantities are specifically referring to some scf workflow (whether it is stored in workflow or not).

Reply:
You mean defining another level to contain the outputs which have required some sort of SCF process? Something like `SCFOutputs(Outputs)` and not defining the scf steps here in the `Outputs` level?

Reply:
Yeah, something like that. I am just wondering what Class I am going to use for my data, I don't think these scf quantities are relevant to me, so if it makes sense to you, I would like to use a higher-level class that doesn't include them

Reply:
Yes, I change the naming then, and you can use `Outputs` (the current `BaseOutputs`) directly